dockstore-webservice/src/main/java/io/dockstore/webservice/FullWorkflowPath.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2022 OICR and UCSC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.dockstore.webservice;
+
+import io.dockstore.common.SourceControl;
+import io.dockstore.webservice.core.SourceControlConverter;
+import io.swagger.annotations.ApiModelProperty;
+import java.io.Serializable;
+import javax.persistence.Column;
+import javax.persistence.Convert;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Table;
+
+/**
+ * Workflow is broken into three different subclasses: BioWorkflow, Service, and AppTool. All three are in separate tables in our database and there is a unique
+ * constraint within each table to prevent duplicate names. However, we want names to be unique across workflows and apptools, so this is being added to keep track
+ * of the full path names within a single table and have the constraint be enforced at the database level.
+ *
+ * @author Natalie
+ */
+
+@Entity
+@Table(name = "fullworkflowpath")
+public class FullWorkflowPath implements Serializable {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
```

Comment:
This is a generated id, but you explicitly set the id in the trigger. I think you probably don't want this.